python/rikai/types/geometry.py

```diff
@@ -281,6 +281,36 @@ def _area(arr: np.ndarray) -> np.array:
             0, arr[:, 3] - arr[:, 1]
         )
 
+    @staticmethod
+    def iou_matrix(box_list1: Union[Sequence[Box2d], np.ndarray],
+                   box_list2: Union[Sequence[Box2d], np.ndarray]) -> np.ndarray:
+        """Compute intersection over union(IOU).
+           For two lists of box2ds, which have the length of N, and M respectively,
+            this function should return a N*M matrix, each element is the iou value (float,[0, 1]).
+        """
+        assert isinstance(box_list1, (Sequence, np.ndarray))
+        assert isinstance(box_list2, (Sequence, np.ndarray))
+
+        if not isinstance(box_list1, np.ndarray):
+            box_list1 = np.array(box_list1)
+        if not isinstance(box_list2, np.ndarray):
+            box_list2 = np.array(box_list2)
+        row_count = box_list1.shape[0]
+        area1 = Box2d._area(box_list1).reshape(row_count, -1)
+        area2 = Box2d._area(box_list2)
+
+        xmin = np.maximum(box_list1[:, 0].reshape((row_count, -1)), box_list2[:, 0])
+        ymin = np.maximum(box_list1[:, 1].reshape((row_count, -1)), box_list2[:, 1])
+        xmax = np.minimum(box_list1[:, 2].reshape((row_count, -1)), box_list2[:, 2])
+        ymax = np.minimum(box_list1[:, 3].reshape((row_count, -1)), box_list2[:, 3])
+
+        inter_area = np.maximum(0, xmax - xmin) * np.maximum(
+            0, ymax - ymin)
+
+        iou_mat = inter_area / (area1 + area2 - inter_area)
```

Comment:
do we need to handle `DivideByZero` here?

Reply:
The default behavior is to print a runtime warning and gives `Nan` in the corresponding cell as a result, I believe it's an ok solution.